File: app/src/main/java/com/example/expense_management/Database/DataItems.java


package com.example.expense_management.Database;


import androidx.room.Entity;
import androidx.room.Ignore;
import androidx.room.PrimaryKey;


import java.util.List;

@Entity(tableName = "expenses")
public class DataItems {


    @PrimaryKey
    private long date;

    private int grossMoneyExpense;

    private int grossMoneyGot;

    private List<Integer> moneyExpense;

    private List<Integer> moneyGot;

    private List<String> moneyGotPurposes;

    private List<String> moneyExpensePurposes;

    public DataItems(){
    }
    @Ignore
    public DataItems(long date, int grossMoneyExpense, int grossMoneyGot, List<Integer> moneyExpense, List<Integer> moneyGot, List<String> moneyGotPurposes, List<String> moneyExpensePurposes) {
        this.date = date;
        this.grossMoneyExpense = grossMoneyExpense;
        this.grossMoneyGot = grossMoneyGot;
        this.moneyExpense = moneyExpense;
        this.moneyGot = moneyGot;
        this.moneyGotPurposes = moneyGotPurposes;
        this.moneyExpensePurposes = moneyExpensePurposes;
    }

    public long getDate() {
        return date;
    }

    public void setDate(long date) {
        this.date = date;
    }

    public int getGrossMoneyExpense() {
        return grossMoneyExpense;
    }

    public void setGrossMoneyExpense(int grossMoneyExpense) {
        this.grossMoneyExpense = grossMoneyExpense;
    }

    public int getGrossMoneyGot() {
        return grossMoneyGot;
    }

    public void setGrossMoneyGot(int grossMoneyGot) {
        this.grossMoneyGot = grossMoneyGot;
    }

    public List<Integer> getMoneyExpense() {
        return moneyExpense;
    }

    public void setMoneyExpense(List<Integer> moneyExpense) {
        this.moneyExpense = moneyExpense;
    }

    public List<Integer> getMoneyGot() {
        return moneyGot;
    }

    public void setMoneyGot(List<Integer> moneyGot) {
        this.moneyGot = moneyGot;
    }

    public List<String> getMoneyGotPurposes() {
        return moneyGotPurposes;
    }

    public void setMoneyGotPurposes(List<String> moneyGotPurposes) {
        this.moneyGotPurposes = moneyGotPurposes;
    }

    public List<String> getMoneyExpensePurposes() {
        return moneyExpensePurposes;
    }

    public void setMoneyExpensePurposes(List<String> moneyExpensePurposes) {
        this.moneyExpensePurposes = moneyExpensePurposes;
    }
}

File: app/src/main/java/com/example/expense_management/database/DataItems.kt
package com.example.expense_management.database

import androidx.room.Entity
import androidx.room.Ignore
import androidx.room.PrimaryKey

@Entity(tableName = "expenses")
class DataItems {
    @PrimaryKey
    var date: Long = 0
    var grossMoneyExpense = 0
    var grossMoneyGot = 0
    var moneyExpense: ArrayList<Int>? =null
    var moneyGot: ArrayList<Int>? = null
    var moneyGotPurposes: ArrayList<String>? = null
    var moneyExpensePurposes: ArrayList<String>? = null

    constructor() {}

    @Ignore
    constructor(
        date: Long,
        grossMoneyExpense: Int,
        grossMoneyGot: Int,
        moneyExpense: ArrayList<Int>?,
        moneyGot: ArrayList<Int>?,
        moneyGotPurposes: ArrayList<String>?,
        moneyExpensePurposes: ArrayList<String>?
    ) {
        this.date = date
        this.grossMoneyExpense = grossMoneyExpense
        this.grossMoneyGot = grossMoneyGot
        this.moneyExpense = moneyExpense
        this.moneyGot = moneyGot
        this.moneyGotPurposes = moneyGotPurposes
        this.moneyExpensePurposes = moneyExpensePurposes
    }
}